File: DSA LeetCode/1514-minimum-value-to-get-positive-step-by-step-sum/solution.py
class Solution:
    def minStartValue(self, nums: List[int]) -> int:
        start_value = 0
        min_value = 0
        for index in range(len(nums)):
            start_value += nums[index]
            min_value = min(min_value, start_value)

        start_value = 1 - min_value
        
        return start_value

File: DSA LeetCode/1514-minimum-value-to-get-positive-step-by-step-sum/solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minStartValue = function(nums) {
    const pre = [nums[0]];
    let minNum = pre[0];
    
    for(let i =1; i<nums.length; i++) {
        pre.push(pre[pre.length -1] + nums[i]);
        minNum = Math.min(minNum, pre[pre.length -1]);
    }
    
    return minNum <= 0 ? Math.abs(minNum) +1 : 1;
};
